cartographer/mapping/internal/work_queue.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_MAPPING_INTERNAL_WORK_QUEUE_H
+#define CARTOGRAPHER_MAPPING_INTERNAL_WORK_QUEUE_H
+
+#include <chrono>
+#include <deque>
+#include <functional>
+
+#include "cartographer/common/time.h"
+
+namespace cartographer {
+namespace mapping {
+
+struct WorkItem {
+  std::chrono::steady_clock::time_point time;
```

Comment:
use common::Time

Reply:
I think `std::chrono::steady_clock` is correct here.

The situation is similar to `kLocalSlamLatencyMetric` and would become very convoluted using sensor timing since it is not readily available at all the necessary places. It is also unclear whether performance should be measured in sensor time during bag playback.

Reply:
Ok, fine with me.